src/factories/Factory.ts

```diff
@@ -1,12 +1,20 @@
 import type { LogisticType } from '@/recipes/logistics/LogisticTypes';
 
+export type FactoryProgressStatus =
+  | 'draft'
+  | 'to_be_done'
+  | 'in_progress'
+  | 'done';
+
 export interface Factory {
   id: string;
-  name?: string | null;
-  description?: string | null;
+  name?: string;
+  description?: string;
```

Comment:
Please keep all undefined values as nullables too, since this make it easier to catch bugs.

Reply:
I'd argue that it's actually the opposite, as a wider state representation leads to more edge cases and reduced typing effectivity.
Also, I don't see the point of the semantic of passing null if you can undefined.

Anyways, it's your code so I won't argue with it.

Reply:
I'm sorry, I worded it badly. I mean that those fields are handled usually through form inputs, and sometimes we're forced to reset to `null` (instead of `undefined`) to avoid some issues with JS where form handlers (through libraries or validators etc.) ignoring the undefined values. It's not something that happened in this project iirc, but I keep the null to avoid nasty bugs - even if this should not happen usually